code/socialDistribution/forms.py

```diff
@@ -83,4 +133,27 @@ def clean_visibility(self):
         if data in [visibility[0] for visibility in LocalPost.Visibility.choices]:
             return data
         else:
-            raise ValidationError('Invalid visibility')
\ No newline at end of file
+            raise ValidationError('Invalid visibility')
+
+    def get_content_and_type(self):
+        """
+            Gets the content type and the content of the submitted data
+        """
+        post_type = self.cleaned_data.get('post_type')
+        image, image_binary, content_type = self.cleaned_data.get('image')
+        
+        # Upload image to media folder and base64-
+        # encode the data to store in database
+        if post_type == PostForm.IMAGE and image:
+            Image.objects.create(
+                caption=self.cleaned_data.get('description'),
+                image=image
```

Comment:
Why do we create an Image here? Do we still need the Image model?

Reply:
I'm using it to upload the images to a media folder like Jawad was suggesting during one of the previous meetings. I thought this was the agreed-upon functionality for #41 . 🤔

Reply:
I thought we were storing the images as binary in the database (`content` field), not the media folder. I just don't see where the media folder is being used? But if there is a reason for it then that's fine too (as long as private posts are still private).

Reply:
I'll ask Jawad for further clarification. But, the only reason I see to keep it is it allows images not related to any post to be embedded in a post being created. Like with ![coolbears](http://127.0.0.1:8000/media/image/coolbears.jpg)

Reply:
Jawad clarified that part with me. You're right, we can the Image model. I made the changes in the latest commits and also moved the endpoint for GETting images from unlisted posts to the socialDistribution/views.py. See unlisted_post_image in socialDistribution/views.py. I'm still trying to figure out why the JS code isn't working properly but I'm unsure if I can get it fixed today.

Reply:
Thanks for the feedback! ♥